### ISSUE TITLE
[Bug] Flash Timeout

### ISSUE THREAD

**Issue by Xanadu16** [jpgohlke/twitch-chat-filter#164]
Whenever the plugin is enabled and I attempt to refresh to connect to the chatroom with the plugin enabled in Greasemonkey, the chat -always- timeouts.

To test this, I turned off the script, but not greasemonkey. Works.
I turned off Greasemonkey. Works.
Greasemonkey is on and the script is too. Does not work.

In the console, it says: 

The connection to wss://s-dal5-nss-16.firebaseio.com/.ws?v=5&ns=ttv-2&s=G1ZFEndGH5rfZEijoQUsMTLhEbA0pem3 was interrupted while the page was loading. global-3e58fa70dd7b74f02d550cbdb2e00a53.js:27
ReferenceError: $ is not defined twitchplayspokemon:1149
"JQMIGRATE: Logging is active" global-3e58fa70dd7b74f02d550cbdb2e00a53.js:17
"JQMIGRATE: jQuery.browser is deprecated" global-3e58fa70dd7b74f02d550cbdb2e00a53.js:17
TypeError: App.Room is undefined twitchplayspokemon:1123
ReferenceError: $ is not defined twitchplayspokemon:1149
ReferenceError: $ is not defined twitchplayspokemon:1149
"build: 0bdb20dfe1205630cef56e664c3134a1de7c60d5" twitchplayspokemon
"TMI.js [conn] ERROR: Could not load the Flash socket SWF. Timed out before receiving the 'loaded' event." tmi-v3.js:1

**Comment by hugomg** [comment]
The "$ is not defined" errors are a bit strange ($ seems to be defined when I tested it) but apparently App.Room really isn't there anymore. Maybe Twitch rewrote the internals of their chat system and if that is the case this could be the end of Twitch Chat Filter :(

**Comment by Xanadu16** [comment]
http://www.reddit.com/r/twitchplayspokemon/comments/2vh6ix/tpp_chat_filter_fix/

**Comment by MattiasBuelens** [comment]
@Xanadu16 Thanks for reporting! Your fix is a good start, I chose to get the `Room` prototype from the defining module instead:
```js
var Room_proto = require("web-client/models/room")["default"].prototype;
```
This should work as long as the Twitch developers don't change the module name - but that would throw a nice, descriptive error in the console so we'd notice it quickly.

I'm working on a more complete fix in #165, but I still have some issues with it. Could you give [this version](https://github.com/MattiasBuelens/twitch-chat-filter/blob/b5baca646ea406e52c29eb50946545428284346d/chat_filter.user.js) a try and let me know how it goes by commenting on that pull request?

**Comment by Xanadu16** [comment]
seems a bit buggy, I switch chat rooms and come back a bit later, when reloading the chat it doesnt apply the filter?
